nx-X11/programs/Xserver/hw/nxagent/Extensions.c

```diff
@@ -338,18 +338,29 @@ static int nxagentRandRInitSizes(ScreenPtr pScreen)
     width = w[i];
     height = h[i];
 
-    mmWidth  = (width * 254 + monitorResolution * 5) / (monitorResolution * 10);
-
-    if (mmWidth < 1)
+    if (monitorResolution < 0)
     {
-      mmWidth = 1;
+      mmWidth  = width * DisplayWidthMM(nxagentDisplay, DefaultScreen(nxagentDisplay)) /
+                     DisplayWidth(nxagentDisplay, DefaultScreen(nxagentDisplay));
+
+      mmHeight = height * DisplayHeightMM(nxagentDisplay, DefaultScreen(nxagentDisplay)) /
+                     DisplayHeight(nxagentDisplay, DefaultScreen(nxagentDisplay));
     }
+    else
+    {
+      mmWidth  = (width * 254 + monitorResolution * 5) / (monitorResolution * 10);
 
-    mmHeight = (height * 254 + monitorResolution * 5) / (monitorResolution * 10);
+      if (mmWidth < 1)
```

Comment:
Move that block and the `if (mmHeight < 1)` block out of the else branch so that it's executed in all cases.

Reply:
> Move that block and the if (mmHeight < 1) block out of the else branch so that it's executed in all cases.

Good point. Done.